pkg/operator/ceph/csi/spec.go

```diff
@@ -711,23 +759,27 @@ func (r *ReconcileCSI) stopDrivers(ver *version.Info) error {
 	return nil
 }
 
-func (r *ReconcileCSI) deleteCSIDriverResources(ver *version.Info, daemonset, deployment, service, driverName string) error {
-	csiDriverobj := v1CsiDriver{}
-	err := k8sutil.DeleteDaemonset(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, daemonset)
-	if err != nil {
-		return errors.Wrapf(err, "failed to delete the %q", daemonset)
+func (r *ReconcileCSI) deleteCSIDriverResources(ver *version.Info, daemonset, deployment, service, driverName string, ownerInfo *k8sutil.OwnerInfo) error {
+	if r.daemonOwnedByRook(ownerInfo, daemonset) {
+		err := k8sutil.DeleteDaemonset(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, daemonset)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete the %q", daemonset)
+		}
 	}
 
-	err = k8sutil.DeleteDeployment(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, deployment)
-	if err != nil {
-		return errors.Wrapf(err, "failed to delete the %q", deployment)
+	if r.provisionerOwnedByRook(ownerInfo, deployment) {
+		err := k8sutil.DeleteDeployment(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, deployment)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete the %q", deployment)
+		}
 	}
 
-	err = k8sutil.DeleteService(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, service)
+	err := k8sutil.DeleteService(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, service)
```

Comment:
still buggy for below code, if they are deployed by someone Rook will delete it

https://github.com/rook/rook/blob/ea8728b7f892518e58fc1f1e36263793997799d9/pkg/operator/ceph/csi/spec.go#L777-L787

Reply:
I can check for `provisionerOwnedByRook` and `dameonOwnedByRook`  if they both are true then only remove the below objects/resources(DeleteService,deleteCSIDriverInfo).

Then I don't see this as an edge case,  
> if they are deployed by someone Rook will delete it

Because I don't see when it can happen somebody else deploys the provisioner and daemon externally and these are deployed by rook.

Reply:
what if someone created a service /csi driver object with the names you are deleting above?

Reply:
If someone is creating those, the provisioner and daemon should be also created externally, a common pattern